pxtblocks/blocklyimporter.ts

```diff
@@ -35,21 +33,28 @@ namespace pxt.blocks {
         return newBlockIds.filter(id => !!workspace.getBlockById(id));
     }
 
-    function applyMetaComments(workspace: Blockly.Workspace) {
+    function applyMetaComments(workspace: Blockly.Workspace, opts?: DomToWorkspaceOptions) {
         // process meta comments
         // @highlight -> highlight block
         workspace.getAllBlocks(false)
             .filter(b => !!b.getCommentText())
             .forEach(b => {
-                const c = b.getCommentText();
-                if (/@highlight/.test(c)) {
-                    const cc = c.replace(/@highlight/g, '').trim();
-                    b.setCommentText(cc || null);
+                const initialCommentText = b.getCommentText();
+                let newCommentText = initialCommentText;
+                if (/@highlight/.test(newCommentText)) {
+                    newCommentText = newCommentText.replace(/@highlight/g, '').trim();
                     (workspace as Blockly.WorkspaceSvg).highlightBlock?.(b.id, true)
                 }
-                if (/@hide/.test(c)) {
+                if (/@hide/.test(newCommentText) && opts.applyHideMetaComment) {
```

Comment:
Could there ever be a case where there is both the `@hide` comment and the `@collapsed` comment on the same block? Would this cause a problem for the `if collapsed` or the following condition if we dispose of the block?

Reply:
There would never be a meaningful situation where they would both apply (as the end result of hide & collapsed would just be hide), but it's a fair point / I'll move the hide to the bottom just in case (it wouldn't ever throw but better safe than sorry / etc)

Reply:
So hiding a block makes it so it doesn't make sense to apply the other flags because we get rid of it, am I understanding that correctly?

Reply:
hiding a block is basicaly equivalent to just deleting it (in fact, that's exactly what the code is doing with the call to dispose, and why it's only enabled with the flag when the blocks are being decompiled for screenshots) - so highlighting, collapsing, or really any other mutation to a block that is going to be deleted doesn't make too much sense, though we might come up with a reason to fiddle with it when we add more flags in the future :)